src/sidebar/reducers/selection.js

```diff
@@ -132,6 +133,17 @@ var update = {
     };
   },
 
+  ADD_ANNOTATIONS(state, action) {
+    var counts = tabs.counts(action.annotations);
+    // If there are no annotations at all, ADD_ANNOTATIONS will not be called.
```

Comment:
I'm not sure about this actually. I don't think anything prohibits this action from being dispatched with an empty list of actions. I think we probably ought to be defensive here and bail if the list of annotations is empty.

Reply:
It is not called if there are no annotations.